sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -674,6 +676,23 @@ class DatasetSuite extends QueryTest with SharedSQLContext {
     }.getMessage
     assert(message.contains("The 0th field of input row cannot be null"))
   }
+
+  test("createTempView") {
+    val dataset = Seq(1, 2, 3).toDS()
+    dataset.createOrReplaceTempView("tempView")
+
+    noException should be thrownBy {
+      // Overrrides the existing temporary view with same name
+      dataset.createOrReplaceTempView("tempView")
+    }
```

Comment:
In general, we try to avoid using fancy ScalaTest assertion facilities like this one. In this case, a simple call to `createOrReplaceTempView` and a comment saying that no exception should be thrown is enough.